lxd/warnings.go

```diff
@@ -502,18 +503,13 @@ func pruneResolvedWarnings(ctx context.Context, s *state.State) error {
 
 // getWarningEntityURL fetches the entity corresponding to the warning from the database, and generates a URL.
 func getWarningEntityURL(ctx context.Context, tx *sql.Tx, warning *cluster.Warning) (string, error) {
```

Comment:
There are two different methods for getting the URL of a LXD entity by it's type and ID.
- `getWarningEntityURL` in `lxd/warnings.go` which accepts a `*sql.Tx` but doesn't cover all entity types.
- `(*Cluster).GetURIFromEntity` in `lxd/db/entity.go` which handles it's own transactions.

I can see that `GetURIFromEntity` will be moved to `ClusterTx` in #12754. When this is done, I think the performance of this function can be improved and used in place of `getWarningEntityURL`.

Reply:
sounds good